code.js

```diff
@@ -1,3 +1,14 @@
 function binarySearch(list, element) {
-    return -1;
+    if (list.length < 1) { return -1; }
+    if (list.length == 1) {
+	if (list[0] == element) { return 0; }
+	return -1;
+    }
+    var index = Math.floor(list.length / 2);
+    if (element == list[index]) {
+	if (element == list[index-1]) { return binarySearch(list.slice(0,index), element); } //finding the smallest index for a match.
```

Comment:
That doesn't really do what your comment says. You also have found the element at this point, so there's no need for a recursive call.

Reply:
This portion of code is my attempt at making it pass the checks. It fails in cases like [4,4] where there are multiple matches; it seems like the code.test.js file wants the smallest indexed match, so I'm not quite sure what to do.

I tried many variations of simpler code like this:
```javascript
if (element == list[index]) {
    return index;
}
```
 But it always fails with a counter example of [x,x].

Reply:
You could change the test code...